test/fn-system-tests/exec_runner_status_test.go

```diff
@@ -302,7 +304,12 @@ func TestCustomHealthChecker(t *testing.T) {
 		t.Fatal("Timeout")
 	}
 	shouldCustomHealthCheckerFail = true
-	defer func() { shouldCustomHealthCheckerFail = false }()
+	defer func() {
+		// Reset test state
+		// Ensure status cache expires
+		shouldCustomHealthCheckerFail = false
+		time.Sleep(2 * time.Second)
```

Comment:
minor: sleep should also check ctx cancel. Maybe use select{} to block on both?

Reply:
Not sure I understand the reason for checking the context cancellation in this last defer block.  The intent here is that this test doesn't leave any state regardless of what happened on the other end, so as to not impact the any other tests when the whole suite is run.  Or are you saying that if ctx was cancelled  then there is a guarantee that the server would've not cached any state?

Reply:
oh, I see the intent now. 2 sec expiry or delay itself is not robust, but this is OK for time being. FWIW, the select stmt above this change doesn't guarantee a 2 sec delay either. Perhaps we could convert this test to poll the status (eg. every 1 sec) until non-cached results are fetched. In general, any test calling Status() should handle and explicitly check for cached results.

Reply:
Right, that would be better.  There are other tests in the same file that use the same (unreliable) pattern of waiting for 2 secs in their setup phase to ensure a clean cache.  I think what we want is a  fixture type that sets ups and cleans the state(including cache unless we are testing caching behavior of course) correctly.  And then refactor all these tests to use that fixture.  We should do that as a separate PR though.